common/authen.js

```diff
@@ -8,16 +8,40 @@
         // authn API no longer communicates through ermrest, removing the need to check for ermrest location
         var serviceURL = $window.location.origin;
 
+        // Private variable to store current session object
+        var _session = null;
+
+        var _changeCbs = [];
+
         return {
 
             getSession: function() {
                 return $http.get(serviceURL + "/authn/session").then(function(response) {
+                    _session = response.data;
```

Comment:
I have removed the` setSessionValue` function. In addition I have made some more changes to the way we fetch session in each app. 

I have introduced a pub sub mechanism, where the recordedit, record and recordset apps listen to session change event once in the beginning and then stop listening it to avoid fetching it twice by authen and app.